### PR TITLE
Initialize new Rust project

### PR DESCRIPTION
**Pull request by Stratus3D**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "asdf-core-rust"
+version = "0.1.0"
+edition = "2021"
+
+description = "Extendable version manager with support for Ruby, Node.js, Elixir, Erlang & more - now in Rust!"
```
Comment (Stratus3D) [inline Cargo.toml:6]: Copied from the Github interface (with an addition). Not sure this will actually get displayed anywhere.

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+homepage = "https://asdf-vm.com/"
+license = "MIT"
+keywords = ["version-manager", "asdf-vm", "multiple-languages", "shell"]
+categories = ["command-line-interface", "command-line-utilities", "config", "development-tools"]
```
Comment (Stratus3D) [inline Cargo.toml:10]: I also added these for good measure, but I think they are only used by crates.io, which I'm not sure we will be using.

File: Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[[bin]]
+name = "asdf"
```
Comment (Stratus3D) [inline Cargo.toml:15]: Make the default binary name `asdf` instead of `asdf-core-rust`.

**Comment by Stratus3D** [comment]
Opened on wrong repo :(

**Comment by hyperupcall** [comment]
:0 Is this code public? I'd like to track it

**Comment by Stratus3D** [comment]
@hyperupcall not yet, as you can see we're just getting started. It will be made public BEFORE it replaces anything in this repo. We will need a lot of folks to test the new Rust implementation before we would feel comfortable replacing the current Bash implementation.

I opened this PR here by accident because I had my local Git remotes setup wrong for the private fork.

**Comment by hyperupcall** [comment]
I see - sounds cool

**Comment by hyperupcall** [comment]
Yes, see more information at https://github.com/asdf-vm/asdf/issues/1565, https://github.com/version-manager/documents, or the rtx project. They explain things in more detail

**Pull request closed without merging**